plugin/src/main/kotlin/de/triplet/gradle/play/internal/PlayPublishPackageBase.kt

```diff
@@ -0,0 +1,52 @@
+package de.triplet.gradle.play.internal
+
+import com.google.api.services.androidpublisher.AndroidPublisher
+import com.google.api.services.androidpublisher.model.LocalizedText
+import com.google.api.services.androidpublisher.model.Track
+import com.google.api.services.androidpublisher.model.TrackRelease
+import java.io.File
+
+abstract class PlayPublishPackageBase : PlayPublishTaskBase() {
+    protected fun AndroidPublisher.Edits.updateTracks(
+            editId: String,
+            inputFolder: File,
+            releaseStatus: String,
+            versions: List<Long>,
+            trackType: String,
+            userPercent: Double
+    ) {
+        val track = tracks()
+                .list(variant.applicationId, editId)
+                .execute().tracks?.firstOrNull { it.track == trackType } ?: Track()
+
+        val releaseText = if (inputFolder.exists()) {
+            inputFolder.listFiles(LocaleFileFilter).mapNotNull { locale ->
+                val fileName = ListingDetail.WHATS_NEW.fileName
+                val file = run {
+                    var file = File(locale, "$fileName-${extension.track}").orNull()
+                    if (file == null) file = File(locale, fileName).orNull()
+                    file
+                } ?: return@mapNotNull null
+                val recentChanges = File(file, fileName).readProcessed(
+                        ListingDetail.WHATS_NEW.maxLength,
+                        extension.errorOnSizeLimit
+                )
+                LocalizedText().setLanguage(locale.name).setText(recentChanges)
+            }
+        } else {
+            null
+        }
+        val trackRelease = TrackRelease().apply {
+            releaseNotes = releaseText
+            status = releaseStatus
+            userFraction = if (status == ReleaseStatus.INPROGRESS.status) userPercent else 0.0
```

Comment:
does this mean users have to set both `userFraction` _and_ `releaseStatus` in the `play` block? That would be a breaking change we would have to explain in a migration doc

Reply:
I guess I'm not understanding. It should have the same functionality as today. If the status isn't set, it defaults to `completed`, if the status is set to `rollout`, then the `userPercent` is used. Neither is required to be set by the user.

Reply:
We have three parameters now: track, status and userPercent. 1.x only had track and userPercent. So current projects that use stayed rollouts are configured like this:

- track: rollout
- userPercent: e.g. 0.05
- no value for status because it didn't exist, thus: defaults to completed

That means we would silently override their userPercent to 0

Reply:
I see what you're saying now. From the API docs. We need to send either `inProgress` or `halted` if we set the release to `rollout`. How about if `track` is `rollout`, then we default to `inProgress`, with overrides still available, and an error if they set `rollout` and an incompatible `status`.